### PR TITLE
Fix failing mute till test

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #16452 

status: ready

### CODE REVIEW

File: src/utils/datetime_test.cljs [hunk 1]
```diff
@@ -190,76 +193,75 @@
...
-         will-unmute-tomorrow      (write-date mock-tomorrow)
-         will-unmute-in-two-days   (write-date mock-in-two-days)
-         will-unmute-in-three-days (write-date mock-in-three-days)
-         will-unmute-in-four-days  (write-date mock-in-four-days)
-         will-unmute-in-five-days  (write-date mock-in-five-days)
-         will-unmute-in-six-days   (write-date mock-in-six-days)))
-      (testing "Until the user turns it back on"
-        (is (= "you turn it back on" (datetime/format-mute-till datetime/go-default-time))))))
+(deftest format-mute-till-test
+    (with-redefs [t/now (constantly (time-coerce/from-long mock-current-time-epoch))]
```
Comment (ilmotta) [inline src/utils/datetime_test.cljs]: @ibrkhalil, I think the proper way is to redef the dynamic var `*ms-fn*` and not the core functions exposed by cljs-time. As we can see, this var is used throughout the cljs-time code https://github.com/andrewmcveigh/cljs-time/blob/master/src/cljs_time/core.cljs#L289. This is also how other tests in `datetime-test` do.

The dynamic var expects a function that returns a number in milliseconds, so you can skip `from-long` and do `(with-redefs [t/*ms-fn* (constantly mock-current-time-epoch)])`

File: src/utils/datetime_test.cljs [hunk 2]
```diff
@@ -190,76 +193,75 @@
...
-        (are [arg expected] (= (datetime/format-mute-till arg) expected)
-         will-unmute-tomorrow      (write-date mock-tomorrow)
-         will-unmute-in-two-days   (write-date mock-in-two-days)
-         will-unmute-in-three-days (write-date mock-in-three-days)
-         will-unmute-in-four-days  (write-date mock-in-four-days)
-         will-unmute-in-five-days  (write-date mock-in-five-days)
-         will-unmute-in-six-days   (write-date mock-in-six-days)))
-      (testing "Until the user turns it back on"
-        (is (= "you turn it back on" (datetime/format-mute-till datetime/go-default-time))))))
+(deftest format-mute-till-test
```
Comment (J-Son89) [inline src/utils/datetime_test.cljs]: imo it seems like `format-mute-till` is quite application specific and this code is in a general `datetime` utilities. Is there a better location to have this code? 
cc @flexsurfer

File: src/utils/datetime_test.cljs [hunk 3]
```diff
@@ -190,76 +193,75 @@
...
-          today-date                #(str (get-hh-mm %) " today")
-          tomorrow-date             #(str (get-hh-mm %) " tomorrow")
-          write-date                #(cond (today? % curr-time)    (today-date %)
-                                           (tomorrow? % curr-time) (tomorrow-date %)
-                                           :else                   (form-full-date %))
-          will-unmute-in-1-hour     (remove-msecs (time-coerce/to-string mock-in-1-hour))
-          will-unmute-in-8-hours    (remove-msecs (time-coerce/to-string mock-in-8-hour))
-          will-unmute-in-15-mins    (remove-msecs (time-coerce/to-string mock-in-15-minutes))
-          will-unmute-in-two-days   (remove-msecs (time-coerce/to-string mock-in-two-days))
-          will-unmute-tomorrow      (remove-msecs (time-coerce/to-string mock-tomorrow))
```
Comment (ilmotta) [inline src/utils/datetime_test.cljs:246]: @ibrkhalil, my suggestion here is not to be necessarily applied in this PR. Unit tests should be easy to read and the expected values should be often hardcoded.

If we set the current cljs-time to a fixed timestamp, then I believe all expected values can be transformed to simple strings and devs will instantly know what this function does and what it produces as a result of any input -> output. I can tell you, I had to start the REPL to know what `format-mute-till` produces.

File: src/utils/datetime_test.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/utils/datetime_test.cljs]: I think it'd be nicer if we have a muting_tests.cljs which has the muting specfic tests.
But for this case, It's a single test and it interacts with time functions so I think it's okish to have it here.

File: src/utils/datetime_test.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/utils/datetime_test.cljs:246]: Understood, I don't think it's worth it to compute the expected values if we already know them, And like you just said it adds extra complexity in understanding what the test does.
I am back Monday, Will create a follow up for the refactor.

Comment (ibrkhalil) [reply src/utils/datetime_test.cljs:246]: https://github.com/status-im/status-mobile/issues/16458

File: src/utils/datetime_test.cljs (hunk 2 above)
Comment (J-Son89) [reply src/utils/datetime_test.cljs]: I was referring to the functionality not to the test itself. It seems like this functionality is misplaced and ofc the test should sit alongside it 👍

Comment (J-Son89) [reply src/utils/datetime_test.cljs]: e.g src/contexts/common/muting seems more applicable imo

**Review by ilmotta — APPROVED** [review]

**Comment by churik** [comment]
@ibrkhalil 
no manual testing is required, so you can merge it

File: src/utils/datetime_test.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/utils/datetime_test.cljs]: Apologies for long response, Had this task on hold for a bit
Addressed here 351b87d9e335008028ad7382629f304646a4185e

**Pull request merged**